examples/factors/README.md

```diff
@@ -26,101 +27,128 @@ cd wherever/you/want
+
+
+In order to access guest code from the host driver program, the host program `host/src/main.rs` includes two guest methods: `METHOD_NAME_ELF` and `METHOD_NAME_ID`.
+
+```no_compile
+use factors_methods::{METHOD_NAME_ELF, METHOD_NAME_ID};
 
-```rust
-use methods::{METHOD_NAME_ELF, METHOD_NAME_ID};
 ```
 
 Both of these must be changed to reflect the new guest program name:
-```rust
-use methods::{MULTIPLY_ELF, MULTIPLY_ID};
+
+```
+use factors_methods::{MULTIPLY_ELF, MULTIPLY_ID};
 ```
-(As an aside, if you add more than one callable guest program to your next RISC Zero zkVM project, you'll need to include these `ELF` and `ID` references once for each guest file.)
+(As an aside, if you add more than one callable guest program to your next RISC Zero zkVM project, you'll need to include `ELF` and `ID` references once for each guest file.)
 
-While we're at it, let's change the rest of the references in `factors/host/src/main.rs`.
-Don't worry about why these lines are included yet; for now, we're just being diligent not to leave dead references behind.
-Here are what the other two lines with `METHOD_NAME_ELF` and `METHOD_NAME_ID` should look like after updating:
+While we're at it, let's update the references to `METHOD_NAME_ELF` and `METHOD_NAME_ID` in `factors/host/src/main.rs` to reflect our new program name. Don't worry about why these lines are included yet; for now, we're just being diligent not to leave dead references behind.
 
-```rust
-    let mut prover =
-        Prover::new(MULTIPLY_ELF).expect("Prover should be constructed from valid ELF binary");
+Here are what the other two lines with `METHOD_NAME_ELF` and `METHOD_NAME_ID` should look like after updating:
 
-...
+```no_compile
+    // Rename METHOD_NAME_ELF
+    let mut exec = Executor::from_elf(env, MULTIPLY_ELF).unwrap();
 
-    receipt.verify(&MULTIPLY_ID).expect(
-        "Code you have proven should successfully verify; did you specify the correct image ID?",
-    );
+    // Rename METHOD_NAME_ID
+    // receipt.verify(MULTIPLY_ID).unwrap();
 ```
 
-### Intermission: Build and run the project!
+### Step 4: Build and run the project!
 
 In the main project folder, build and run the project using `cargo run --release`.
 Nothing exciting will happen yet, but it should give you an error if you made one of the above changes incorrectly.
 Use this command any time you'd like to check your progress.
 
-## Step 3 (Host): Share two values with the guest
+## Concept break: How do we run and prove the guest program?
+
+Our next objective is to provide the guest program with input. Before we implement this, let's take a closer look at how we run and prove the guest program in `factors/src/main.rs`.
+
+In the starter template project, our host driver program creates an executor environment before constructing an executor.  When the `Executor::run()` command executes, it will produce a session. The `Session::prove()` command will cause our guest program to execute, producing a receipt:
```

Comment:
```suggestion
In the starter template project, our host driver program creates an executor environment before constructing an executor.  When `Executor::run()` is called, it will produce a session. Calling `Session::prove()` will cause our guest program to execute, producing a receipt:
```